### ISSUE TITLE
Spectrogram of LH or VH channels

### ISSUE THREAD

**Issue by sbonaime** [obspy/obspy#1411]
ObsPy version 1.0.1, MacOS 10.11.5,Python 2.7.11 Anaconda 2.3.0

Hi,

I think there is a bug with the spectrogram function with low frequency sampling files. In the following example, I get a spectrogram for the BHZ but not the LHZ channel

``` python

from obspy.core import UTCDateTime
from obspy.clients.arclink import Client

client = Client(user="test_spectrogram@ipgp.fr")

start = UTCDateTime("2013-02-07 21:35:00")
end = start + 86400

st = client.get_waveforms("G", "IVI", "00", "BHZ", start, end)
print st
st.spectrogram()

st = client.get_waveforms("G", "IVI", "00", "LHZ", start, end)
print st
st.spectrogram()
```

the error is

``` log

  File "/Users/bonaime/obspy/spectroblem.py", line 19, in <module>
    st.spectrogram()

  File "/Users/bonaime/git/obspy/obspy/core/stream.py", line 1161, in spectrogram
    spec = tr.spectrogram(**kwargs)

  File "/Users/bonaime/git/obspy/obspy/core/trace.py", line 863, in spectrogram
    return spectrogram(data=self.data, **kwargs)

  File "/Users/bonaime/git/obspy/obspy/imaging/spectrogram.py", line 133, in spectrogram
    pad_to=mult, noverlap=nlap)

  File "/Users/bonaime/anaconda/lib/python2.7/site-packages/matplotlib/mlab.py", line 1325, in specgram
    mode=mode)

  File "/Users/bonaime/anaconda/lib/python2.7/site-packages/matplotlib/mlab.py", line 743, in _spectral_helper
    result = stride_windows(x, NFFT, noverlap, axis=0)

  File "/Users/bonaime/anaconda/lib/python2.7/site-packages/matplotlib/mlab.py", line 567, in stride_windows
    raise ValueError('noverlap must be less than n')

ValueError: noverlap must be less than n
```

**Comment by legovini** [comment]
While this bug gets fixed, is there any known workaround?

**Comment by megies** [comment]
You can work around this by not relying on the default parameters in http://docs.obspy.org/master/packages/autogen/obspy.imaging.spectrogram.spectrogram.html

Looking at the source http://docs.obspy.org/master/_modules/obspy/imaging/spectrogram.html#spectrogram you should be able to work around this by adjusting `per_lap`..

```python
st.spectrogram(per_lap=...)
```

**Comment by paride** [comment]
Unfortunately with a 1 sps trace it doesn't seem to work with any value of `per_lap`.

**Comment by megies** [comment]
well, `per_lap` is not the only parameter you should tweak..?

**Comment by ThomasLecocq** [comment]
Hey,
I took 5 minutes to read the doc and test:

this works smooth:

```python
st.spectrogram(wlen=60)
st.spectrogram(wlen=360)
st.spectrogram(wlen=3600)
```

because, if not provided, ``wlen`` defaults to ``tr.stats.sampling_rate / 100.0`` ; which is not documented, but you can read it in the code:

https://docs.obspy.org/_modules/obspy/imaging/spectrogram.html#spectrogram

@megies: want me to PR a doc update? And, yay, the default ``sps/100.0`` is a bit hardcoded to obtain 1 second for 100 Hz data, or ? Shouldn't it, somehow, prevent or warn if the calculated ``wlen`` is smaller than (n times) ``delta``?

**Comment by megies** [comment]
@ThomasLecocq Go ahead. Docs improvements are always very welcome.

Default for `wlen` probably is really not ideal, I guess something like `tr.stats.delta * 1e3` or so would be suited better for a wider variety of sampling rates.. would need a deprecation step across one major version if we change though.

Still, I think doing a spectrogram, the user should understand what is going on and take control of those vital parameters.

**Comment by megies** [comment]
This was finally fixed in #3093
